build.sbt

```diff
@@ -29,21 +40,20 @@ lazy val core = project.in(file("core")).settings(scalaModuleSettings).settings(
   )
 )
 
-lazy val junit = project.in(file("junit")).settings(
+lazy val junit = project.in(file("junit")).settings(disablePublishing).settings(
```

Comment:
Minor/pro-tip: No need to jump in and out of `.settings`.

TL;DR: you could define this as

```scala
project.in(file("junit")).settings(
  disablePublishing,
  libraryDependencies += "com.novocode" % "junit-interface" % "0.11" % "test",
```

---

Long-winded explanation:

Individual settings (type `Setting[_]`) and seq of settings (type `Seq[Setting[_]]`) are both `SettingDefinition`s (the former by definition, the latter via implicit lift), and `.settings` is defined to take a varargs of `SettingDefinition`, so they can be defined next to each other.

For convenience there is also a `Def.settings(ss: SettingsDefinition*): Seq[Setting[_]]`